lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp

```diff
@@ -850,4 +850,110 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
             binder.op, resultType, tensor, /*memory_format=*/noneVal);
         return success();
       });
+  patterns.onOp(
+      "Mean", 1, [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+        if (binder.op->getNumOperands() == 1) {
+          Torch::ValueTensorType resultType;
+          Value x;
+          if (binder.tensorOperand(x) || binder.tensorResultType(resultType))
+            return failure();
+          rewriter.replaceOp(binder.op, x);
+          return success();
+        }
+        Torch::ValueTensorType resultType;
+        SmallVector<Value> valList;
+        int64_t numOperands = binder.op->getNumOperands();
+        Value numOperandsConstant = rewriter.create<Torch::ConstantIntOp>(
+            binder.getLoc(), rewriter.getType<Torch::IntType>(),
+            rewriter.getIntegerAttr(rewriter.getIntegerType(64), numOperands));
+        if (binder.tensorOperands(valList, numOperands) ||
+            binder.tensorResultType(resultType))
+          return failure();
+        Value const1 = rewriter.create<Torch::ConstantIntOp>(
+            binder.getLoc(), rewriter.getType<Torch::IntType>(),
+            rewriter.getIntegerAttr(rewriter.getIntegerType(64), 1));
+        // Short circuit to binary add
+        if (numOperands == 2) {
+          Value sum = rewriter.create<Torch::AtenAddTensorOp>(
+              binder.getLoc(), resultType, valList[0], valList[1], const1);
```

Comment:
You can keep this out of this if condition and as a result this value can be used in the later part of code in place of `curr`.

Reply:
Here we use resultType, but later in the code we use baseType so need both currently.

Reply:
Hi @saienduri, I don't see any difference between 'Value sum' at line no. 947, and the `Value curr` at line no. 957. That's why I said that we can keep only only the first one by taking it out of the if condition.

Reply:
ah makes sense, done